src/wireviz/wv_dataclasses.py

```diff
@@ -0,0 +1,815 @@
+    ignore_in_bom: bool = False
+    bom_id: Optional[str] = None  # to be filled after harness is built
+
+    def fill_partnumbers(self):
+        partnos = [self.pn, self.manufacturer, self.mpn, self.supplier, self.spn]
+        partnos = [remove_links(entry) for entry in partnos]
+        partnos = tuple(partnos)
+        self.partnumbers = PartNumberInfo(*partnos)
+
+    def parse_number_and_unit(
+        self,
+        inp: Optional[Union[NumberAndUnit, float, int, str]],
+        default_unit: Optional[str] = None,
+    ) -> Optional[NumberAndUnit]:
+        if inp is None:
+            return None
+        elif isinstance(inp, NumberAndUnit):
+            return inp
+        elif isinstance(inp, float) or isinstance(inp, int):
+            return NumberAndUnit(float(inp), default_unit)
+        elif isinstance(inp, str):
+            if " " in inp:
+                number, unit = inp.split(" ", 1)
+            else:
+                number, unit = inp, default_unit
+            try:
+                number = float(number)
+            except ValueError:
+                raise Exception(
+                    f"{inp} is not a valid number and unit.\n"
+                    "It must be a number, or a number and unit separated by a space."
+                )
+            else:
+                return NumberAndUnit(number, unit)
+
+    @property
+    def bom_hash(self) -> BomHash:
+        if self.sum_amounts_in_bom:
+            _hash = BomHash(
+                description=self.description,
+                qty_unit=self.amount.unit if self.amount else None,
+                amount=None,
+                partnumbers=self.partnumbers,
+            )
+        else:
+            _hash = BomHash(
+                description=self.description,
+                qty_unit=self.qty.unit,
+                amount=self.amount,
+                partnumbers=self.partnumbers,
+            )
+        return _hash
+
+    @property
+    def bom_qty(self) -> float:
+        if self.sum_amounts_in_bom:
+            if self.amount:
+                return self.qty.number * self.amount.number
+            else:
+                return self.qty.number
+        else:
+            return self.qty.number
+
+    def bom_amount(self) -> NumberAndUnit:
+        if self.sum_amounts_in_bom:
+            return NumberAndUnit(None, None)
+        else:
+            return self.amount
+
+    @property
+    def has_pn_info(self) -> bool:
+        return any([self.pn, self.manufacturer, self.mpn, self.supplier, self.spn])
+
+
+@dataclass
+class AdditionalComponent(Component):
+    qty_multiplier: Union[QtyMultiplierConnector, QtyMultiplierCable, int] = 1
+    _qty_multiplier_computed: Union[int, float] = 1
+    designators: Optional[str] = None  # used for components definedi in the
+    #                                    additional_bom_items section within another component
+    bgcolor: SingleColor = None  #       ^ same here
+    note: str = None
```

Comment:
Why use the same class for both `additional_bom_items` and `additional_components`? None of the members in this class seems to be common for these two usages (`designators` only in the former usage, and the rest only in the latter usage)?

What are the disadvantages with defining `class AdditionalBomItems(Component)` and class AdditionalComponent(GraphicalComponent) as separate classes?

Reply:
No real reason comes to mind. Feel free to split.